### PR TITLE
Add support for reCaptcha v3

### PR DESCRIPTION
**Pull request by ajibarra**

### CODE REVIEW

File: src/View/Helper/UserHelper.php [hunk 1]
```diff
@@ -167,11 +167,26 @@ public function addReCaptcha()
...
+        throw new \InvalidArgumentException(__d('cake_d_c/users', 'reCaptcha version is wrong. Please configure Users.reCaptcha.version as 2 or 3'));
+
+    }
 
+    /**
+     * Add required element for reCaptcha v2
+     *
+     * @return string
+     */
+    private function addReCaptchaV2()
```
Comment (steinkel) [inline src/View/Helper/UserHelper.php]: let's add a deprecation warning in case of V2 and switch default to V3 in the next major version?

File: templates/Users/register.php [hunk 2]
```diff
@@ -35,6 +35,6 @@
         }
         ?>
     </fieldset>
-    <?= $this->Form->button(__d('cake_d_c/users', 'Submit')) ?>
+    <?= $this->User->button(__d('cake_d_c/users', 'Submit')) ?>
```
Comment (steinkel) [inline templates/Users/register.php:38]: why this change?

**Review by steinkel — CHANGES_REQUESTED** [review]

File: templates/Users/register.php (hunk 2 above)
Comment (ajibarra) [reply templates/Users/register.php:38]: You can see https://github.com/CakeDC/users/pull/1085/files#diff-627db50119981d9f17e65e3606d2a3e45c1292417ee73a2a88b5e5ed07cc9cceR214. Recaptcha V3 requires some properties to be added to the button

File: src/View/Helper/UserHelper.php [hunk 3]
```diff
@@ -42,7 +44,7 @@ class UserHelper extends Helper
      * @param array $options options
      * @return string
      */
-    public function socialLogin($name, $options = [])
+    public function socialLogin(string $name, array $options = []): string
```
Comment (steinkel) [inline src/View/Helper/UserHelper.php:47]: adding return types qualify for breaking BC, so it should go to branch 15.next

File: src/View/Helper/UserHelper.php [hunk 4]
```diff
@@ -167,10 +169,29 @@ public function addReCaptcha()
...
         $this->addReCaptchaScript();
-        try {
-            $this->Form->unlockField('g-recaptcha-response');
-        } catch (\Exception $e) {
+        $version = Configure::read('Users.reCaptcha.version', 2);
+        $method = "addReCaptchaV$version";
+        if (method_exists($this, $method)) {
+            try {
+                $this->Form->unlockField('g-recaptcha-response');
+            } catch (Exception $e) {
```
Comment (steinkel) [inline src/View/Helper/UserHelper.php:177]: could we check for existing Form helper instead of silent capture the exception here?

**Review by steinkel — CHANGES_REQUESTED** [review]
